packages/flutter_tools/test/commands.shard/permeable/build_bundle_test.dart

```diff
@@ -559,6 +559,42 @@ void main() {
     ProcessManager: () => FakeProcessManager.any(),
   });
 
+  testUsingContext('values from --dart-define supersede values from --dart-define-from-file', () async {
+    globals.fs
+        .file(globals.fs.path.join('lib', 'main.dart'))
+        .createSync(recursive: true);
+    globals.fs.file('pubspec.yaml').createSync();
+    globals.fs.file('.packages').createSync();
+    globals.fs.file('.env').writeAsStringSync('''
+        MY_VALUE=VALUE_FROM_ENV_FILE
+      ''');
+    final CommandRunner<void> runner =
+        createTestCommandRunner(BuildBundleCommand(
+      logger: BufferLogger.test(),
+    ));
+
+    await runner.run(<String>[
+      'bundle',
+      '--no-pub',
+      '--dart-define=MY_VALUE=VALUE_FROM_COMMAND',
+      '--dart-define-from-file=.env',
+    ]);
+
+  }, overrides: <Type, Generator>{
+    BuildSystem: () => TestBuildSystem.all(BuildResult(success: true),
+            (Target target, Environment environment) {
+          expect(
+            _decodeDartDefines(environment),
+            containsAllInOrder(const <String>[
+              'MY_VALUE=VALUE_FROM_ENV_FILE',
+              'MY_VALUE=VALUE_FROM_COMMAND',
```

Comment:
so if the dart vm sees these, the second entry will override the first?

Reply:
Yeah, I tested this on MacOS using the repro steps (I should have pushed my repro project, but I accidentally `git clean`ed it out of existence).

Reply:
I would have liked to write a more sound test for this, but I am not sure we can do this without actually doing a real build.

Alternatively, we _could_ implement parsing on `--dart-define` rebuild our own string that removes redundant entries, but this would be nontrivial.

Reply:
yeah, if that's non-trivial this is fine. just wanted to make sure i was understanding correctly.